DigitalLearningSolutions.Web/Controllers/NotificationPreferencesController.cs

```diff
@@ -34,5 +33,35 @@ public IActionResult Index()
 
             return View(model);
         }
+
+        [Route("/NotificationPreferences/Edit/{userType}")]
+        public IActionResult UpdateNotificationPreferences(string userType)
+        {
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Login");
+            }
+
+            IEnumerable<NotificationPreference> notifications;
+
+            if (userType == UserTypes.Admin)
+            {
+                var adminId = User.GetCustomClaimAsInt(CustomClaimTypes.UserAdminId);
+                notifications = notificationPreferencesDataService.GetNotificationPreferencesForAdmin(adminId);
+            }
+            else if (userType == UserTypes.Delegate)
+            {
+                var delegateId = User.GetCustomClaimAsInt(CustomClaimTypes.LearnCandidateId);
+                notifications = notificationPreferencesDataService.GetNotificationPreferencesForDelegate(delegateId);

```

Comment:
It would be nice to refactor this into it's own method (e.g. GetNotificationPreferencesForUser that checks for user type and handles this logic).

Reply:
I'll need to move the class (or enum) that distinguishes the user types into the Data project (unless we put the method in the controller), where would be a good location? Or should it be handled in the Web project?